src/olympia/blocklist/admin.py

```diff
@@ -0,0 +1,211 @@
+
+
+@admin.register(Block)
+class BlockAdmin(BlockAdminAddMixin, admin.ModelAdmin):
+    list_display = (
+        'guid',
+        'min_version',
+        'max_version',
+        'updated_by',
+        'modified')
+    readonly_fields = (
+        'addon_guid',
+        'addon_name',
+        'addon_updated',
+        'users',
+        'review_listed',
+        'review_unlisted',
+    )
+    ordering = ['-modified']
+    view_on_site = False
+    list_select_related = ('updated_by',)
+    actions = ['delete_selected']
+
+    addon_instance = None
+
+    class Media:
+        css = {
+            'all': ('css/admin/blocklist_block.css',)
+        }
+
+    def addon_guid(self, obj):
+        return self.addon_instance.guid
+    addon_guid.short_description = 'Add-on GUID'
+
+    def addon_name(self, obj):
+        return self.addon_instance.name
+
+    def addon_updated(self, obj):
+        return self.addon_instance.modified
+
+    def users(self, obj):
+        return self.addon_instance.average_daily_users
+
+    def review_listed(self, obj):
+        has_listed = any(
+            True for v in self._get_addon_versions().values()
+            if v == amo.RELEASE_CHANNEL_LISTED)
+        if has_listed:
+            url = reverse(
+                'reviewers.review',
+                kwargs={'addon_id': self.addon_instance.pk})
+            return format_html(
+                '<a href="{}">{}</a>', url, _('Review Listed'))
+        return ''
+
+    def review_unlisted(self, obj):
+        has_unlisted = any(
+            True for v in self._get_addon_versions().values()
+            if v == amo.RELEASE_CHANNEL_UNLISTED)
+        if has_unlisted:
+            url = reverse(
+                'reviewers.review', args=('unlisted'),
+                kwargs={'addon_id': self.addon_instance.pk})
+            return format_html(
+                '<a href="{}">{}</a>', url, _('Review Unlisted'))
+        return ''
+
+    def get_fieldsets(self, request, obj):
+        return (
+            (None, {
+                'fields': (
+                    'addon_guid',
+                    'addon_name',
+                    'addon_updated',
+                    'users',
+                    ('review_listed', 'review_unlisted'))
+            }),
+            ('Add New Block' if not obj else 'Edit Block', {
+                'fields': (
+                    'min_version',
+                    'max_version',
+                    'url',
+                    'reason',
+                    'include_in_legacy'),
+            }),
+        )
+
+    def save_model(self, request, obj, form, change):
+        obj.updated_by = request.user
+        if not change:
+            obj.addon = self.addon_instance
+        super().save_model(request, obj, form, change)
+
+    def get_queryset(self, request):
+        return super().get_queryset(request).prefetch_related(
+            Prefetch(
+                'addon', queryset=Addon.objects.all().only_translations()),
```

Comment:
`Addon.unfiltered`, not `Addon.objects`, otherwise that will not work with deleted add-ons. I made this mistake in abuse report admin, see https://github.com/mozilla/addons-server/issues/12742